runs/restore.sh

```diff
@@ -1,20 +1,31 @@
 #!/bin/bash
-echo "****************************************"
-echo "Step 1: moving file to home directory..."
-sudo cp /var/www/html/openWB/web/tools/upload/backup.tar.gz /home/pi/backup.tar.gz
-cd /home/pi/
-echo "****************************************"
-echo "Step 2: extracting archive..."
-sudo tar -vxf backup.tar.gz
-echo "****************************************"
-echo "Step 3: replacing old files..."
-sudo cp -v -R /home/pi/var/www/html/openWB/* /var/www/html/openWB/
-sudo chmod 777 /var/www/html/openWB/openwb.conf
-echo "****************************************"
-echo "Step 4: cleanup after restore..."
-sudo rm /var/www/html/openWB/web/tools/upload/backup.tar.gz
-sudo rm /home/pi/backup.tar.gz
-sudo rm -R /home/pi/var
-echo "****************************************"
-echo "End: Restore finished."
-echo "****************************************"
+OPENWBBASEDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+SOURCEFILE="$OPENWBBASEDIR/web/tools/upload/backup.tar.gz"
+WORKINGDIR="/home/pi/openwb_restore"
+LOGFILE="$OPENWBBASEDIR/web/tools/upload/restore.log"
+
+{
+	echo "$(date +"%Y-%m-%d %H:%M:%S") Restore of backup started..."
+	echo "****************************************"
+	echo "Step 1: creating working directory \"$WORKINGDIR\""
+	mkdir -p "$WORKINGDIR"
+	echo "****************************************"
+	echo "Step 2: extracting archive to working dir \"$WORKINGDIR\"..."
+	sudo tar -vxf "$SOURCEFILE" -C "$WORKINGDIR"
+	echo "****************************************"
+	echo "Step 3: replacing old files..."
+	cp -v -R -p "${WORKINGDIR}${OPENWBBASEDIR}/." "${OPENWBBASEDIR}R/"
```

Comment:
Wieso sind das zwei Schritte? Warum nicht einfach
```bash
sudo tar -vxf "$SOURCEFILE" -C /
```
(mosquitto müsste dann natürlich auch vorher gestoppt...).

Also Notiz am Rande: Das backup stellt Dateien wiederher, nicht den ursprünglichen Zustand: Wenn seit dem Backup neue Dateien dazu gekommen sind, bleiben diese bestehen.

P.S.: Was macht das `R` am Ende von `"${OPENWBBASEDIR}R/"`?

Reply:
> Wieso sind das zwei Schritte? Warum nicht einfach
> sudo tar -vxf "$SOURCEFILE" -C /

Weil man dann sicher ist das das Tar file sich auch absturzfrei auspacken läst.
Ansonsten hast du "mischmasch"

Reply:
Den Mischmasch hast du auch so. Der Return-Code von `tar` wird nicht abgefragt. So wie es aktuell ist hat tar möglicherweise irgendwann den Dienst quittiert. Wenn man das verhindern will wäre es sinnvoll bei einem non-zero return code von tar abzubrechen. Außerdem wäre es sinnvoll `mv` statt `cp` zu nehmen, damit jedes Überschreiben "atomic" ist und nicht möglicherweise nach dem eine Datei zur Hälfte kopiert wurde der Prozess abbricht. Außerdem könnte man beim Einsatz von `mv` den temporär benötigten Speicherplatz reduzieren.

Reply:
Das "R" war noch ein Überbleibsel meiner Tests. Wollte nicht immer meine Installation zerschießen und habe in ein anderes Verzeichnis kopiert. Ist jetzt entfernt.

Ich werte jetzt den Exitstatus von tar aus und breche ab, falls etwas schief gelaufen ist. Außerdem `cp`durch `mv`ersetzt.

Reply:
Das Archiv direkt nach "/" auszupacken halte ich für sehr gewagt. Mit einem manipulierten Archiv kann ich mir dann alles auf der Box installieren, was ich gerne hätte. So werden nur die Dateien überschrieben, die openWB auch erwartet.